src/GitVersion.Core/VersionCalculation/VariableProvider.cs

```diff
@@ -31,6 +31,8 @@ public VersionVariables GetVariablesFor(SemanticVersion semanticVersion, Effecti
                 semanticVersion.PreReleaseTag.Name = config.GetBranchSpecificTag(this.log, semanticVersion.BuildMetaData?.Branch, null);
                 if (semanticVersion.PreReleaseTag.Name.IsNullOrEmpty())
                 {
+                    // TODO: Why do we manipulating the semantic version here in the VariableProvider? The method name is GET not MANIPULATE.
+                    // What is about the separation of concern and single-responsibility principle?
```

Comment:
Agreed. No method named `Get` should do manipulation. I want most classes within `Core` to be immutable. Feel free to refactor (in a separate PR against `support/5.x`). :)

Reply:
I agree the class e.g. NextVersion and SematicVersion should be immutable.